src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -93,86 +101,83 @@ def details_widget
+
+          richtext_checkbox(id:       addon_widget_id(addon),
+                            label:    label,
+                            selected: addon_selected?(addon),
+                            enabled:  addon.available?,
+                            indented: addon.depends_on)
         end
-
-        VWeight(75, MarginBox(2, 1, content))
-      end
-
-      # display the addon checkboxes in two columns
-      # @param col1 [Array<Addon>] the addons displayed in the first column
-      # @param col2 [Array<Addon>] the addons displayed in the second column
-      # @return [Yast::Term] the addon cheboxes
-      def two_column_layout(col1, col2)
-        box2 = addon_selection_items(col1)
-        box2.params << VStretch() # just UI tweak
-
-        HBox(
-          addon_selection_items(col2),
-          HSpacing(1),
-          box2
-        )
+        items.join("\n")
       end
 
-      # create a single UI column with addon checkboxes
-      # @return [Yast::Term] addon column
-      def addon_selection_items(addons)
-        box = VBox()
-
-        # whether to add extra spacing in the UI
-        add_extra_spacing = if Yast::UI.TextMode
-          addons.size < 5
+      IMAGE_DIR = "/usr/share/YaST2/theme/current/wizard".freeze
+      IMAGES = {
+        "normal:on:enabled"   => "checkbox-on.png",
+        "normal:off:enabled"  => "checkbox-off.png",
+        # theme has no special images for disabled checkboxes
+        "normal:on:disabled"  => "checkbox-on.png",
+        "normal:off:disabled" => "checkbox-off.png",
+        "inst:on:enabled"     => "inst_checkbox-on.png",
+        "inst:off:enabled"    => "inst_checkbox-off.png",
+        "inst:on:disabled"    => "inst_checkbox-on-disabled.png",
+        "inst:off:disabled"   => "inst_checkbox-off-disabled.png"
+      }.freeze
+
+      # Make a simulation of a CheckBox displayed in a RichText
+      # @param id [String]
+      # @param label [String]
+      # @param selected [Boolean]
+      # @param enabled [Boolean]
+      # @param indented [Boolean]
+      # @return [String] a Value for a RichText
+      def richtext_checkbox(id:, label:, selected:, enabled:, indented: false)
+        selected = false unless enabled
+        if Yast::UI.TextMode
+          indent = "&nbsp;" * (indented ? 5 : 1)
+          check = selected ? "[x]" : "[ ]"
+          widget = "#{check} #{label}"
+          enabled_widget = enabled ? "<a href=\"#{id}\">#{widget}</a>" : widget
+          "#{indent}#{enabled_widget}<br>"
         else
-          true
-        end
+          indent = "&nbsp;" * (indented ? 7 : 1)
 
-        addons.each do |addon|
-          box.params.concat(addon_checkbox(addon, add_extra_spacing))
-        end
+          installation = !Yast::Mode.normal
+          image = (installation ? "inst:" : "normal:") +
+            (selected ? "on:" : "off:") + (enabled ? "enabled" : "disabled")
+          color = installation ? "white" : "black"
```

Comment:
Yes, it is fragile. I have discussed this with Ken. We thought of using qss classes, but since this is a RichText, it uses yet another stylesheet, and it will not work out.